consumer_topic.go

```diff
@@ -11,15 +11,20 @@ type subscriptionEvent struct {
 type Topic struct {
 	Name         string
 	lagTolerance int64
-	partitionLag map[int32]int64
+	partitionLag map[int32]partitionLagState
```

Comment:
Not requiring a change here because it's a personal preference but I'd just change this one to `map[int32]*int64`.

The thought process behind it is that I can't imagine another unhandled case for the lag. We'd have either:
- positive lag when the log end offset is greater than or equal to the consumer one (99.99% of the cases); here we use two valid values
- negative lag when something is totally broken in the response of the sarama APIs (sanity check, it's something I never expect to happen but we handle it properly); here we use two valid values leading to an unexpected result
- nil lag when the consumer one is not committed (0.01% of the cases, applicable to newly started consumers where no new messages are produced in the topic for some time); here we use one valid and one empty value so the calculation doesn't make sense

It's basically aiming to get rid of lag wrapping structures and it could definitely feel like some magic interpretation of the pointer so I'm good with the current solution as well.